### ISSUE TITLE
Have you considering using mavlink instead of direct PWM manipulation?

### ISSUE THREAD

**Issue by DanielSolmann** [AnishDey27/Object-Tracking-Drone#1]
There is an article in ardupilot wiki: https://ardupilot.org/dev/docs/raspberry-pi-via-mavlink.html
And you can use DroneKit library for python: https://dronekit-python.readthedocs.io/en/latest/guide/quick_start.html

**Comment by AnishDey27** [comment]
Yup we can use Mavlink for Apm and pixhawk only but to make it compitable with all opensource FCs we have to use separate arduino for pwm handling.

**Comment by DanielSolmann** [comment]
Can you, please, give me an example of such incompatible FC? I generally agree with you, but it's not FC, which determines protocol availability, but its firmware.

**Comment by sampathkumarch** [comment]
yes ,  i also got same doubt , uploading in  rpi code there  there is no dronekit api

**Comment by Tirupatibalaji-Dev** [comment]
> Can you, please, give me an example of such incompatible FC? I generally agree with you, but it's not FC, which determines protocol availability, but its firmware.
you can use pymavlink to make autonomous drone
You can refer [this](https://www.ardusub.com/developers/pymavlink.html)

**Comment by Tirupatibalaji-Dev** [comment]
> Yup we can use Mavlink for Apm and pixhawk only but to make it compitable with all opensource FCs we have to use separate arduino for pwm handling.

hey can you tell me which tracker package you installed?

**Comment by AnishDey27** [comment]
> > Yup we can use Mavlink for Apm and pixhawk only but to make it compitable with all opensource FCs we have to use separate arduino for pwm handling.
> 
> hey can you tell me which tracker package you installed?



> > Yup we can use Mavlink for Apm and pixhawk only but to make it compitable with all opensource FCs we have to use separate arduino for pwm handling.
> 
> hey can you tell me which tracker package you installed?



> > Yup we can use Mavlink for Apm and pixhawk only but to make it compitable with all opensource FCs we have to use separate arduino for pwm handling.
> 
> hey can you tell me which tracker package you installed?

Centroid Tracker

**Comment by Tirupatibalaji-Dev** [comment]
Leave i made my own tracker ai
